carp.client.core/src/commonMain/kotlin/dk/cachet/carp/client/domain/StudyRuntime.kt

```diff
@@ -108,24 +117,51 @@ class StudyRuntime private constructor(
      *
      * @return True in case deployment succeeded; false in case device could not yet be deployed (e.g., awaiting registration of other devices).
      * @throws UnsupportedOperationException in case deployment failed since not all necessary plugins to execute the study are available.
+     * @throws IllegalStateException in case data requested in the deployment cannot be collected on this client.
      */
-    suspend fun tryDeployment( deploymentService: DeploymentService ): Boolean
+    suspend fun tryDeployment( deploymentService: DeploymentService, dataCollector: DataCollector ): Boolean
     {
         val deploymentStatus = deploymentService.getStudyDeploymentStatus( studyDeploymentId )
-        val clientDeviceStatus = deploymentStatus.getDeviceStatus( device )
 
-        return tryDeployment( deploymentService, clientDeviceStatus )
+        return tryDeployment( deploymentService, dataCollector, deploymentStatus )
     }
 
-    private suspend fun tryDeployment( deploymentService: DeploymentService, deviceStatus: DeviceDeploymentStatus ): Boolean
+    private suspend fun tryDeployment(
+        deploymentService: DeploymentService,
+        dataCollector: DataCollector,
+        deploymentStatus: StudyDeploymentStatus
+    ): Boolean
     {
         // Early out in case state indicates the device is not yet ready to deploy.
+        val deviceStatus = deploymentStatus.getDeviceStatus( device )
         if ( deviceStatus !is DeviceDeploymentStatus.NotDeployed || !deviceStatus.isReadyForDeployment ) return false
 
-        // TODO: Get deployment and throw exception in case there are missing plugins to perform the operations (e.g., measurements).
+        // Get deployment information.
         val deployment = deploymentService.getDeviceDeploymentFor( studyDeploymentId, device.roleName )
+        check( deployment.deviceDescriptor == device )
         deploymentInformation = deployment
 
+        // Verify whether data can be collected for all requested measures.
+        for ( deviceTasks in deployment.getTasksPerDevice() )
+        {
+            val dataTypes = deviceTasks.tasks.flatMap { it.measures }.map { it.type }.distinct()
+            for ( type in dataTypes )
+            {
+                val canCollectData =
+                    if ( deviceTasks.isConnectedDevice )
+                    {
+                        dataCollector.canCollectDataForConnectedDevice(
+                            type,
+                            deviceTasks.device::class,
+                            deviceTasks.deviceRegistration )
+                    }
+                    else dataCollector.canCollectData( type )
+
+                check( canCollectData )
```

Comment:
Does `canCollectDataForConnectedDevice` try to connect to the external sensor? We cannot always guarantee connection to an external sensor at this point. The external sensor might be out of reach. If the external sensor is out of reach the caller of `tryDeployment` cannot determine which external sensor the client cannot collect data from. This could be vital information to communicate to the user of the client.

Reply:
No. It only determines whether the runtime has the necessary integrations loaded to connect to a sensor of that type with the given device registration.

Reply:
`supportsDataCollection( DataType )` and `supportsDataCollectionForConnectedDevice`?

Reply:
Fixed in https://github.com/cph-cachet/carp.core-kotlin/pull/165/commits/088ff54924c56034bb573bb40e1d85204619e5a2